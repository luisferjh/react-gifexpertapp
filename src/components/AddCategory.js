import React, {useState} from 'react'
import PropTypes from 'prop-types'

export const AddCategory = ({setCategories}) => {

    const [inputValue, setInputValue] = useState('')

    const handInputChange = (e) => {
        setInputValue(e.target.value)          
        
        console.log('handleInputChange llamado')
    }

    const handleSubmit = (e)=>{
        e.preventDefault();            
        if(inputValue.trim().length > 2)
        {
            setCategories(category => [inputValue, ...category])
            setInputValue('')
        }

    }

    return (
        <>
        <form onSubmit={handleSubmit}>
            <p>{inputValue}</p>
            <input
                type="text"
                value={inputValue}
                onChange={handInputChange}
            />
        </form>
            
        </>
    )
}

AddCategory.propTypes = {
    setCategories:PropTypes.func.isRequired
}
