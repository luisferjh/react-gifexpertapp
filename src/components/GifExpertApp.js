import React, {useState} from 'react'
import {AddCategory} from './AddCategory'
import {GifGrid} from './GifGrid'

export const GifExpertApp = () => {

   // const categories = ['One Puch', 'samurai X', 'dragon ball']
    const [categories, setCategories] = useState(['samurai X'])

    // const handleAdd = ()=>{
    //     console.log(categories)
    //     //setCategories([...categories, 'Hunter x Hunter']) // alternativa para no borrar el estado
    //     setCategories(cats =>  [...cats, 'Hunter x Hunter']) // alternativa para no borrar el estado
    // }


    console.log(categories)
    return (
        <>
            <h2>GifExpertApp</h2>
            <AddCategory setCategories={setCategories} />
            <hr />        

            <ol>
               {
                   categories.map((category, index) => ( // los parentesis indican que se devuelve un objeto
                    <GifGrid 
                        key={category}
                        category={category} />
                   ))
               }
            </ol>
           
        </>
    )
}
