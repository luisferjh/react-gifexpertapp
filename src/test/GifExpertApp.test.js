import React from 'react'
import '@testing-library/jest-dom'
import { shallow } from 'enzyme'
import { GifExpertApp } from '../components/GifExpertApp'

describe('Debe mostrarse correctamente', ()=>{

    
    
    test('debe de mostrarse correctamente', async () => {
        const wrapper = shallow(<GifExpertApp /> )
        expect(wrapper).toMatchSnapshot();
    })

    test('debe de mostrar una lista de categorias', ()=> {

        const categories = ['samurai x', 'dragon ball']
        const wrapper = shallow(<GifExpertApp defaultCategories={categories} /> )

        expect(wrapper).toMatchSnapshot();
        expect(wrapper.find('GifGrid').length).toBe(categories.length);
    })
})