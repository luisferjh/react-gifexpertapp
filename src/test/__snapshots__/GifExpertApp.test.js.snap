// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Debe mostrarse correctamente debe de mostrar una lista de categorias 1`] = `
<Fragment>
  <h2>
    GifExpertApp
  </h2>
  <AddCategory
    setCategories={[Function]}
  />
  <hr />
  <ol>
    <GifGrid
      category="samurai x"
      key="samurai x"
    />
    <GifGrid
      category="dragon ball"
      key="dragon ball"
    />
  </ol>
</Fragment>
`;

exports[`Debe mostrarse correctamente debe de mostrarse correctamente 1`] = `
<Fragment>
  <h2>
    GifExpertApp
  </h2>
  <AddCategory
    setCategories={[Function]}
  />
  <hr />
  <ol />
</Fragment>
`;
